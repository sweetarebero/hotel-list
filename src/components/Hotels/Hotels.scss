.Hotels {
  .row {
    width: 100%;
  }
  .col {
    padding: 0;
  }
  .col:last-of-type > * {
    margin-left: auto;
  }
}
