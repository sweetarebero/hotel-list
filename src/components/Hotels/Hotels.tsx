import * as React from "react";
import useAxios from "axios-hooks";
import { Col, Row } from "reactstrap";
import HotelCount from "../HotelCount/HotelCount";
import HotelList from "../HotelList/HotelList";
import SortHotel from "../SortHotels/SortHotel";
import SortProvider from "../SortProvider/SortProvider";

interface Props {}

const Hotels = (props: Props) => {
  const [{ data, loading, error }] = useAxios("http://localhost:4000/results");

  if (loading) return <p>Loading...</p>;
  if (error) return <p>Error!</p>;

  return (
    <SortProvider>
      <div className="sort-section">
        <Row>
          <Col>
            <HotelCount count={data.length} city="Sydney" />
          </Col>
          <Col>
            <SortHotel />
          </Col>
        </Row>
      </div>
      <HotelList hotels={data} />
    </SortProvider>
  );
};

export default Hotels;
