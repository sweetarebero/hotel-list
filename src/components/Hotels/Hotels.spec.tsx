import React from "react";
import { render } from "@testing-library/react";
import Hotels from "./Hotels";
import "@testing-library/jest-dom/extend-expect";

describe("Hotels render", () => {
  it("renders correctly", () => {
    const { container } = render(<Hotels />);
    expect(container).toMatchSnapshot();
  });
});
