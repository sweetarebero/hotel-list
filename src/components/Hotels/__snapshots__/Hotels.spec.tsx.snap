// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Hotels renders correctly 1`] = `
<div>
  <p>
    Loading...
  </p>
</div>
`;
